### PR TITLE
[RESTEASY-2700] - Removed deprecated processor API code

### PR DESCRIPTION
**Pull request by lalifk**

### CODE REVIEW

File: server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ChunkOutputStream.java [hunk 1]
```diff
@@ -33,39 +32,42 @@
...
-    /**
-     * This is used to establish {@link #byteSink} upon the first writing of bytes.
-     */
-   private final Supplier<FluxSink<Tuple2<byte[], CompletableFuture<Void>>>> byteSinkSupplier;
+   /**
+    * This is used to establish {@link #byteSink} upon the first writing of bytes.
+    */
+   private final Supplier<Sinks.Many<Tuple2<byte[], CompletableFuture<Void>>>> byteSinkSupplier;
+
+   private final EmitFailureHandler emitFailureHandler = EmitFailureHandler.FAIL_FAST;
```
Comment (crankydillo) [inline server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ChunkOutputStream.java]: I just noticed that this is member variable.  Did you think about making it a static variable?

File: server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyHttpResponse.java [hunk 2]
```diff
@@ -290,4 +301,16 @@ public void flushBuffer() throws IOException {
         log.trace("Flushing response buffer!");
         out.flush();
     }
+
+    private void completionSinkSubscribe(final Mono<Void> mono) {
```
Comment (crankydillo) [inline server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyHttpResponse.java]: We do the same thing elsewhere and also maintain state to do it.  We could have isolated this.  Not huge deal though.

File: server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java [hunk 3]
```diff
@@ -235,12 +239,15 @@ public int read() {
                    } else {
                       sendMono = resp.status(500).send();
                    }
-                   sendMono.subscribe(completionMono);
+                   sendMono.subscribe(
```
Comment (crankydillo) [inline server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java]: The third place we do this..

File: server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java [hunk 4]
```diff
@@ -91,18 +93,20 @@ public ReactorNettyJaxrsServer start() {
...
 
-      HttpServer svrBuilder = HttpServer
-          .create()
-          .tcpConfiguration(this::configure)
-          .port(configuredPort)
-          .handle(handler::handle);
+      HttpServer svrBuilder =
+              Optional.ofNullable(idleTimeout)
+              .map(to -> HttpServer.create()
+                      .doOnConnection(cc -> mkConfigureConnConsumer().accept(cc)))
```
Comment (crankydillo) [inline server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java]: Can't `mkConfigureConnConsumer` just become a 'normal' Java method (i.e. `configure(Connection)`) and then this just becomes `doOnConnection(ReactorNettyJaxrsServer#configure)` ?

**Review by crankydillo — CHANGES_REQUESTED** [review]
Question about using statics and the one about `Consumer<Connection>` are the main things I want answers to.

File: server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ChunkOutputStream.java (hunk 1 above)
Comment (lalifk) [reply server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ChunkOutputStream.java]: Why would you prefer it to be a static variable in this case?

File: server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java (hunk 4 above)
Comment (lalifk) [reply server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java]: Sure we can do that

Comment (lalifk) [reply server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java]: We can do that

File: server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java (hunk 3 above)
Comment (lalifk) [reply server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java]: First and only place in this class. I created a separate class & method and used it across.

File: server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ChunkOutputStream.java (hunk 1 above)
Comment (crankydillo) [reply server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ChunkOutputStream.java]: To only maintain one pointer in memory as opposed to one per request/response.  Micro-optimization and probably eliminated by the runtime, but I see constants used in much more dubious places:)

File: server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/SinkSubscriber.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package org.jboss.resteasy.plugins.server.reactor.netty;
+
+import java.util.function.Consumer;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Sinks;
+import reactor.core.publisher.Sinks.EmitFailureHandler;
+
+public class SinkSubscriber {
+
```
Comment (crankydillo) [inline server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/SinkSubscriber.java:9]:
```suggestion
private SinkSusbcriber() {
}

```

**Review by crankydillo — APPROVED** [review]
Would be nice to have that private constructor added, but if you are ready to be done with this.  I'm good:)  I will wait about 30mins or so before merging.

**Review by crankydillo — APPROVED** [review]

File: server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/SinkSubscriber.java (hunk 5 above)
Comment (lalifk) [reply server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/SinkSubscriber.java:9]: Just saw it now :-)

**Pull request merged**